sources/api/pluto/src/ec2.rs

```diff
@@ -0,0 +1,76 @@
+use crate::proxy;
+use aws_smithy_types::error::display::DisplayErrorContext;
+use aws_types::region::Region;
+use snafu::{OptionExt, ResultExt, Snafu};
+use std::time::Duration;
+
+// Limit the timeout for the EC2 describe-instances API call to 5 minutes
+const EC2_DESCRIBE_INSTANCES_TIMEOUT: Duration = Duration::from_secs(300);
```

Comment:
Should we align this with the timeout that the in-tree aws cloud provider uses for the same API call?

I am wary about creating a launch time regression for anyone using a private VPC with no access to the EC2 API. If this call timed out previously and the cloud provider just used the system hostname, then it may not have been noticed.

Reply:
The in-tree aws cloud provider relies on the default retries/timeout in the aws-sdk-go EC2 client: https://github.com/kubernetes/legacy-cloud-providers/blob/74c983775777a7ddd7ebcf5973420adb405cefd4/aws/instances.go#L136-L147

They don't modify the default client:
https://github.com/kubernetes/legacy-cloud-providers/blob/74c983775777a7ddd7ebcf5973420adb405cefd4/aws/aws.go#L824-L834

> A default value of 2 for maximum retry attempts, making a total of 3 call attempts. This value can be overwritten through the max_attempts configuration parameter. 
> Any retry attempt will include an exponential backoff by a base factor of 2 for a maximum backoff time of 20 seconds. 

Do we want to align on that? 

EKS Optimized AMI is doing 10 retries instead:
https://github.com/awslabs/amazon-eks-ami/pull/1264/files#diff-049390d14bc3ea2d7882ff0f108e2802ad9b043336c5fa637e93581d9a7fdfc2R488-R490

Reply:
I still think we should put a time bound on the API call still. If we dont, we'll run into https://github.com/bottlerocket-os/bottlerocket/issues/2370 again where `pluto` can potentially block boot for 30 mins (default standard retry configuration).

Reply:
> I am wary about creating a launch time regression for anyone using a private VPC with no access to the EC2 API. If this call timed out previously and the cloud provider just used the system hostname, then it may not have been noticed.

From looking at the legacy cloud provider code, I believe the call would have hanged for longer than 5 mins while blocking `kubelet` from progressing. Every AWS SDK implementation has a different default request timeout value it seems... but most are between 100 seconds to 240 seconds. Which across 3 retries would put it over the 5 min timeout we're enforcing.

 In the EKS optimized AMI, I believe the bootstrapping strip is probably blocking for longer than 5 mins as well with the 10 retries (each retry is 1 mins according to [`aws-cli` docs](https://docs.aws.amazon.com/cli/latest/reference/)).